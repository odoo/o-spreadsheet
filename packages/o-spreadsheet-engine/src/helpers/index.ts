export * from "./misc";
export * from "./numbers";
export * from "./references";
export * from "./uuid";
