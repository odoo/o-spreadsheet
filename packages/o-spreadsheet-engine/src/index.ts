export { FunctionCodeBuilder, Scope, type FunctionCode } from "./formulas/code_builder";
export {
  OPERATOR_MAP,
  UNARY_OPERATOR_MAP,
  compile,
  compileTokens,
  functionCache,
} from "./formulas/compiler";
export {
  OP_PRIORITY,
  convertAstNodes,
  iterateAstNodes,
  mapAst,
  parse,
  parseTokens,
} from "./formulas/parser";
export type {
  AST,
  ASTFuncall,
  ASTOperation,
  ASTString,
  ASTSymbol,
  ASTUnaryOperation,
} from "./formulas/parser";
export { rangeTokenize } from "./formulas/range_tokenizer";
export { POSTFIX_UNARY_OPERATORS, tokenize } from "./formulas/tokenizer";
export type { Token } from "./formulas/tokenizer";
export { FunctionRegistry } from "./functions/function_registry";
export { categories } from "./functions/functionRegistry";
export * from "./helpers";
export { Model } from "./model";
export { BasePlugin } from "./plugins/base_plugin";
export { Registry } from "./registry";
export { StateObserver, type StateObserverChange } from "./state_observer";
export * from "./translation";
export { CellValue } from "./types/cells";
export { SpreadsheetClipboardData } from "./types/clipboard";
export { CoreGetters, PluginGetters } from "./types/coreGetters";
export * from "./types/errors";
export { Format } from "./types/format";
export * from "./types/history";
export * from "./types/locale";
export * from "./types/misc";
export * from "./types/validator";
export const __info__ = {};
