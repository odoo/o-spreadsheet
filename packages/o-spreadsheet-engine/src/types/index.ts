export * from "./base";
export * from "./errors";
export * from "./functions";
export * from "./history";
export * from "./isMatrix";
export * from "./locale";
export * from "./validator";
