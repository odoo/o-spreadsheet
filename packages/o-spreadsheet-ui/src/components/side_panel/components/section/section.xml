<templates>
  <t t-name="o_spreadsheet.Section">
    <div class="o-section" t-att-class="props.class">
      <t t-if="props.slots.title or props.title">
        <div class="o-section-title">
          <t t-esc="props.title"/>
          <t t-slot="title"/>
        </div>
      </t>
      <t t-slot="default"/>
    </div>
  </t>
</templates>
