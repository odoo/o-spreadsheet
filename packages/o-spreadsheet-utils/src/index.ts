export * from "./translation";
export * from "./uuid";
