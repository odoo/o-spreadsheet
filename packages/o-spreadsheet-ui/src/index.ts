import * as Engine from "@odoo/o-spreadsheet-engine";

export function salut() {
  console.log("Salut !");
}

export function createSpreadsheetModel() {
  return Engine.SpreadsheetModel();
}