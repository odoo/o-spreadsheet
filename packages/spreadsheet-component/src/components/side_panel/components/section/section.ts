import { Component } from "@odoo/owl";
import { SpreadsheetChildEnv } from "../../../../types";

interface Props {
  class?: string;
}

export class Section extends Component<Props, SpreadsheetChildEnv> {
  static template = "o_spreadsheet.Section";
  static props = {
    class: { type: String, optional: true },
    slots: Object,
  };
}
