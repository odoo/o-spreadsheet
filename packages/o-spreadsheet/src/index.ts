import * as UI from "@odoo/o-spreadsheet-ui";


export { UI };