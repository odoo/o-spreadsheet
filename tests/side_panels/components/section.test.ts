import { Component, xml } from "@odoo/owl";
import { Section } from "../../../src/components/side_panel/components/section/section";
import { SpreadsheetChildEnv } from "../../../src/types";
import { mountComponent } from "../../test_helpers/helpers";

let fixture: HTMLElement;

type Props = Section["props"];

describe("Section", () => {
  test("Can render a section", async () => {
    class SectionContainer extends Component<Props, SpreadsheetChildEnv> {
      static template = xml/* xml */ `
    <div class="container">
      <Section t-props="props"/>
    </div>
  `;
      static components = { Section };
    }
    const props = { title: "My title", class: "my-class" };
    ({ fixture } = await mountComponent(SectionContainer, { props }));
    expect(fixture).toMatchSnapshot();
  });

  test("Can add a content to a section", async () => {
    class SectionContainer extends Component<Props, SpreadsheetChildEnv> {
      static template = xml/* xml */ `
    <div class="container">
      <Section t-props="props">
        <div class="content">My content</div>
      </Section>
    </div>
  `;
      static components = { Section };
    }
    const props = { title: "My title", class: "my-class" };
    ({ fixture } = await mountComponent(SectionContainer, { props }));
    expect(fixture).toMatchSnapshot();
  });
});
