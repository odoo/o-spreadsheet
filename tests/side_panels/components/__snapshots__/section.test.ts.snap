// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Can render a section with a title 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="o-section my-class"
    >
      <div
        class="o-section-title"
      >
        My title
      </div>
      
      <div
        class="content"
      >
        My content
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Section Can render a section without a title 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="o-section my-class"
    >
      
      <div
        class="content"
      >
        My content
      </div>
      
    </div>
  </div>
</div>
`;
